packages/graphback-codegen-client/src/plugin/templates/gqlTemplates.ts

```diff
@@ -0,0 +1,195 @@
+import { getFieldName, GraphbackOperationType, ModelDefinition } from '@graphback/core'
+import { GraphQLObjectType } from 'graphql';
+import { ClientTemplate } from './ClientTemplates';
+import { buildReturnFields, printReturnFields } from './fragmentFields';
+
+export const fragment = (t: GraphQLObjectType) => {
+  const queryReturnFields = buildReturnFields(t, 0);
+  const returnFieldsString = printReturnFields(queryReturnFields);
+
+  return `fragment ${t.name}Fields on ${t.name} {
+${returnFieldsString}
+} `
+}
+
+// TODO describe fragments in doc CRUD spec
+export const expandedFragment = (t: GraphQLObjectType) => {
+  const queryReturnFields = buildReturnFields(t, 1);
+  const returnFieldsString = printReturnFields(queryReturnFields);
+
+  return `fragment ${t.name}ExpandedField on ${t.name} {
+${returnFieldsString}
+} `
+}
+
+export const findAllQuery = (t: GraphQLObjectType) => {
+  const fieldName = getFieldName(t.name, GraphbackOperationType.FIND_ALL, 's')
+
+  return `query ${fieldName} {
+    ${fieldName} {
+      ...${t.name}ExpandedFields
+    }
+  }`
+}
+
+export const findQuery = (t: GraphQLObjectType) => {
+  const fieldName = getFieldName(t.name, GraphbackOperationType.FIND, 's')
+
+  return `query ${fieldName} ($filter: NoteFilter!)}) {
+    ${fieldName}(filter: $filter}) {
+      ...${ t.name}ExpandedFields
+    }
+  }`
+}
+
+
+export const createMutation = (t: GraphQLObjectType) => {
+  const fieldName = getFieldName(t.name, GraphbackOperationType.CREATE)
+
+  return `mutation ${fieldName} ($input: ${t.name}Input!) {
+  ${ fieldName} (input: ${t.name}Input!) {
+      ...${ t.name}Fields
+  }
+}
+`
+}
+
+export const updateMutation = (t: GraphQLObjectType) => {
+  const fieldName = getFieldName(t.name, GraphbackOperationType.UPDATE)
+
+  return `mutation ${fieldName} ($input: ${t.name}Input!) {
+  ${ fieldName} (input: ${t.name}Input!) {
+      ...${ t.name}Fields
+  }
+}
+`
+}
+
+export const deleteMutation = (t: GraphQLObjectType) => {
+  const fieldName = getFieldName(t.name, GraphbackOperationType.DELETE)
+
+  return `mutation ${fieldName} ($input: ${t.name}Input!) {
+  ${fieldName} (input: ${t.name}Input!) {
+      ...${t.name}Fields
+  }
+}
+`
+}
+
+export const subscription = (t: GraphQLObjectType, subscriptionType: string) => {
+  const fieldName = `${subscriptionType}${t.name} `
+
+  return `subscription ${fieldName} {
+  ${fieldName} {
+      ...${t.name}Fields
+  }
+} `
+}
+
+export const createFragments = (types: ModelDefinition[]) => {
+  return types.reduce((data: ClientTemplate[], model: ModelDefinition) => {
+    data.push({
+      name: model.graphqlType.name,
+      implementation: fragment(model.graphqlType)
+    })
+    data.push({
```

Comment:
What is the purpose of multiple fragment types for a model? Is one for mutations and one (expanded) for queries?